### ISSUE TITLE
[Feature Request] Ability to hide links

### ISSUE THREAD

**Issue by data-flux** [michael-lazar/rtv#326]
Thanks for this amazing program!

I have only one problem with it, I like to hide links, when I've seen them.

As far as I can tell, this is not currently possible.

If at all possible, could you add this feature?

**Comment by data-flux** [comment]
Clarification, this is what I hope to do:

https://www.reddit.com/dev/api/#POST_api_hide

**Comment by michael-lazar** [comment]
It looks like there's a function exposed through PRAW here https://github.com/praw-dev/praw/blob/praw3/praw/__init__.py#L2605 ,so I don't think that this would be very difficult at all. Are you thinking of having a config option to auto-hide links that were visited, or having a key that you can press to hide the selected link?

**Comment by data-flux** [comment]
Keypress for sure.

**Comment by armandg** [comment]
This featur would be great! Any chance of setting this so you could chose auto-hide or hide by keypress? Perhaps keypress also should hide/show the posts you already read?

**Comment by nemanjan00** [comment]
Any news on this?

**Comment by michael-lazar** [comment]
No, there hasn't been any work done on this yet. Since there have been several people commenting / voting on this issue I added a high priority tag.

**Comment by nemanjan00** [comment]
Ok, guys, I am working on PR. 

It would be great if someone could also test it. :)

Thank you!

**Comment by armandg** [comment]
@nemanjan00 If I get some starting help with getting your PR down to my machine and running I will be more than happy to test =)

**Comment by nemanjan00** [comment]
```bash
git clone git@github.com:nemanjan00/rtv.git
cd rtv
git checkout hide
python -m rtv
```

Button for hiding is x. 

If you have custom config, add: 

```
HIDE = x
```

above 

```
UPVOTE = a
```

**Comment by armandg** [comment]
Both standard shortcut and changing in settings works great.

A few things that also should be fixed:

- The "hide" button should toggle hidden or not, just like for saving posts or comments (press `w` multiple times on a post, and you'll see what I mean).
- Should this function perhaps also be described in the bottom status bar with the others?

**Comment by nemanjan00** [comment]
Ok, I will see what I can do.

**Comment by armandg** [comment]
@nemanjan00 Also, trying to hide a comment crashes rtv and gives following output:

```Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/geir/Dropbox/coding/git/rtv/rtv/__main__.py", line 242, in <module>
    sys.exit(main())
  File "/home/geir/Dropbox/coding/git/rtv/rtv/__main__.py", line 225, in main
    page.loop()
  File "rtv/page.py", line 82, in loop
    self.controller.trigger(ch)
  File "rtv/objects.py", line 590, in trigger
    return func(self.instance, *args, **kwargs)
  File "rtv/subreddit_page.py", line 179, in open_submission
    page.loop()
  File "rtv/page.py", line 82, in loop
    self.controller.trigger(ch)
  File "rtv/objects.py", line 590, in trigger
    return func(self.instance, *args, **kwargs)
  File "rtv/page.py", line 31, in wrapped_method
    return f(self, *args, **kwargs)
  File "rtv/page.py", line 163, in hide
    data['object'].hide()
  File "rtv/packages/praw/objects.py", line 92, in __getattr__
    raise AttributeError(msg)
AttributeError: '<class 'rtv.packages.praw.objects.Comment'>' has no attribute 'hide'```

**Comment by nemanjan00** [comment]
Ok. Will fix that now. 

Can you run git pull inside rtv and run it again, to test hide toggle?

**Comment by armandg** [comment]
Toggling works great. Crash when trying to hide a comment still exist (comments can't be hidden IIRC, so it should not work if the item is something else than a submission).

**Comment by nemanjan00** [comment]
Fixed comments problem now.

**Comment by nemanjan00** [comment]
Added footer documentation

**Comment by armandg** [comment]
Commentproblem confirmed fixed.

Statusbar is apparently different when viewing a subreddit and when viewing a post, I get different views.

![Subreddit](https://user-images.githubusercontent.com/265139/34829747-fea74e6c-f6e1-11e7-9679-6d6f5491c873.png)

![Submission](https://user-images.githubusercontent.com/265139/34829759-0abd945e-f6e2-11e7-9b51-0bdd7f323719.png)

**Comment by nemanjan00** [comment]
Yeah, it is. Should it be in second one also?

**Comment by armandg** [comment]
No you are absolutely correct. I'm obviously not thinking clear right now 😂

I'm satisfied with the testing so far =)

**Comment by nemanjan00** [comment]
I do not know python and tbh have no idea what I am doing so I do not know when and if this will be merged.

**Comment by armandg** [comment]
It seems to be stopping when testing in Travis CI. I program small scripts in Python so I also have no idea 😂

Looks like we wait for @michael-lazar to check the incoming PR.

**Comment by nemanjan00** [comment]
Tests finally passed. Big thanks to @michael-lazar for helping me resolve problems!

**Comment by michael-lazar** [comment]
Closing, this was merged in #498
